client/plugins/userLibrary.ts

```diff
@@ -0,0 +1,111 @@
+import {
+  ItemsApiGetItemsRequest,
+  UserLibraryApiGetLatestMediaRequest
+} from '@jellyfin/client-axios';
+import { Plugin } from '@nuxt/types/app';
+
+type GetItemsParams = Omit<ItemsApiGetItemsRequest, 'userId'>;
+type GetLatestMediaParams = Omit<UserLibraryApiGetLatestMediaRequest, 'userId'>;
+
+type UserLibraryType = {
+  getItem: (id: string) => Promise<void>;
+  getItems: (params: GetItemsParams) => Promise<string[]>;
+  getLatestMedia: (params: GetLatestMediaParams) => Promise<string[]>;
+};
+
+declare module '@nuxt/types' {
+  interface Context {
+    $userLibrary: UserLibraryType;
+  }
+
+  interface NuxtAppOptions {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+declare module 'vuex/types/index' {
+  // eslint-disable-next-line -- Current TypeScript rules flag S as unused, but Nuxt requires identical types
+  interface Store<S> {
+    $userLibrary: UserLibraryType;
+  }
+}
+
+const userLibraryPlugin: Plugin = ({ $api, $auth, store }, inject) => {
+  inject('userLibrary', {
+    /**
+     * Fetches an item based on its ID and stores it
+     *
+     * @param {string} id - Item ID
+     */
+    getItem: async (id: string): Promise<void> => {
+      const call = async (): Promise<void> => {
+        const item = (
+          await $api.userLibrary.getItem({
+            userId: $auth.user?.Id,
+            itemId: id
+          })
+        ).data;
+
+        await store.dispatch('items/addItem', { item });
+      };
+      const promise = call();
+
+      if (!store.getters['items/getItem'](id)) {
+        await promise;
+      }
+    },
+
+    /**
+     * Executes a getItems API call with given parameters, stores the result and returns the ID list
+     *
+     * @param {object} params - Parameters of getItems API call (without user ID)
+     * @returns {string[]} list of IDs
+     */
+    getItems: async (params: GetItemsParams): Promise<string[]> => {
+      const result = (
+        await $api.items.getItems({ ...params, userId: $auth.user?.Id })
+      ).data.Items;
+
+      if (!result) {
+        return [];
+      }
+
+      await store.dispatch('items/addItems', { items: result });
+
+      // @ts-expect-error - stupid parser thinking it'll be an (undefined|string)[] when I explicitly filter them
```

Comment:
No first person in comments. Also watch the language :yum: 

A better description would be:
```suggestion
      // @ts-expect-error - The parser fails to infer types properly here.
```